package/python-kiwi-spec-template

```diff
@@ -75,6 +75,23 @@ The KIWI Image System provides an operating system image builder
 for Linux supported hardware platforms as well as for virtualization
 and cloud systems like Xen, KVM, VMware, EC2 and more.
 
+%package -n kiwi-develdeps
+Summary:        KIWI - Development dependencies
+Group:          %{sysgroup}
+Requires:       gcc
+Requires:       glibc-devel
+Requires:       libffi-devel
+Requires:       trang
+Requires:       libxml2-devel
+Requires:       libxslt-devel
+Requires:       python%{python3_pkgversion}-tox
+Requires:       python%{python3_pkgversion}-%{develsuffix} >= 3.6
+Requires:       python%{python3_pkgversion}-setuptools
+
+%description -n kiwi-develdeps
+This metapackage installs the necessary dependencies
+to develop KIWI.
```

Comment:
I don't think this will be useful/necessary once I've gotten #2441 in place.

Reply:
ok so you are saying poetry also cares for the non python based requirements like "trang" to be present ? How does it do that ?

Reply:
Hmm, well, I guess not trang. But isn't trang already a weak dependency we use at runtime?

Reply:
no for development this is a strict dependency. See the Makefile
```
kiwi/schema/kiwi.rng: kiwi/schema/kiwi.rnc
    # whenever the schema is changed this target will convert
    # the short form of the RelaxNG schema to the format used
    # in code and auto generates the python data structures
    trang -I rnc -O rng kiwi/schema/kiwi.rnc kiwi/schema/kiwi.rng
```
It's required to convert the schema to RelaxNG which is the format we use. So if you as a developer do a single schema change you require this tool to convert it into the right format.

I guess not many people know that though ;)

Reply:
If you don't work with the schema you don't need it, but it would be good if we can come up with a concept that provides the full environment not only parts of it

Reply:
That's why I thought it would be handy that we provide this as a package in the same way you did it with the systemdeps packages and reference to it in the documentation

Reply:
gcc is needed for the C tools/ that we still have in the project to support users with custom initrds. But I would say all that C code should be dropped.

In the end outside of python I only see

* trang

The -devel packages are only in there when "pip install ..." starts to compile code from the module it fetches. I never completely understood why it sometimes compiles stuff and sometimes not, maybe poetry fixes that ? but it imho depends on how the module is provided on pypi

Reply:
@Conan-Kudo I'm down to trang as the only package outside of python. I was thinking we could add trang to the systemdeps-core even if that would not be 100% accurate. But if you would agree I can add the one package there and drop the -develdeps. Thoughts ?

Reply:
If we have only one dependency, we could express the check in the Makefile, no? Because trang is packaged nowhere but in openSUSE.

Reply:
ok that can also be done

Reply:
@Conan-Kudo ok done, this should be acceptable now, please check. Thanks